tower-http/src/auth/async_require_authorization.rs

```diff
@@ -204,6 +205,7 @@ where
     fn call(&mut self, req: Request<ReqBody>) -> Self::Future {
         let inner = self.inner.clone();
         let authorize = self.auth.authorize(req);
+        let inner = mem::replace(&mut self.inner, inner);
```

Comment:
Can use `mem::swap`, which is shorter. Also, mind adding a comment about why the service and its clone need swapping?